MediaBrowser.Controller/Entities/BaseItem.cs

```diff
@@ -1417,39 +1413,25 @@ private static Video[] LoadThemeVideos(IEnumerable<FileSystemMetadata> fileSyste
 
         protected virtual BaseItem[] LoadExtras(List<FileSystemMetadata> fileSystemChildren, IDirectoryService directoryService)
         {
-            var extras = new List<Video>();
-
-            var libraryOptions = new LibraryOptions();
-            var folders = fileSystemChildren.Where(i => i.IsDirectory).ToList();
-            foreach (var extraFolderName in AllExtrasTypesFolderNames)
-            {
-                var files = folders
-                    .Where(i => string.Equals(i.Name, extraFolderName, StringComparison.OrdinalIgnoreCase))
-                    .SelectMany(i => FileSystem.GetFiles(i.FullName));
-
-                // Re-using the same instance of LibraryOptions since it looks like it's never being altered.
-                extras.AddRange(LibraryManager.ResolvePaths(files, directoryService, null, libraryOptions)
+            return fileSystemChildren
+                .Where(child => child.IsDirectory)
+                .Where(folder => AllExtrasTypesFolderNames.ContainsKey(folder.Name))
```

Comment:
No need to iterate twice
```suggestion
                .Where(child => child.IsDirectory && AllExtrasTypesFolderNames.ContainsKey(child.Name))
```

Reply:
I called Where() twice for better readability. Internally it won't iterate twice.  See: https://github.com/dotnet/runtime/blob/main/src/libraries/System.Linq/src/System/Linq/Where.SpeedOpt.cs

Reply:
It doesn't seem to be free.

<details><summary>Code</summary>

```
        private static string[] Test =
        {
            "asdasdasdsad1",
            "2sadsadsadsadsa",
            "3dsadsadsadsadsa",
            "3dsadsadsadsadsa",
            "4dsadsadsadsadsa",
            "5dsadsadsadsadsa",
            "1dsadsadsadsadsa",
            "2dsadsadsadsadsa",
            "3dsadsadsadsadsa",
            "4dsadsadsadsadsa",
            "5dsadsadsadsadsa",
            "1dsadsadsadsadsa",
            "2dsadsadsadsadsa",
            "3dsadsadsadsadsa",
            "45dsadsadsadsadsa",
            "53dsadsadsadsadsa",
            "1dsadsadsadsadsa",
            "2dsadsadsadsadsa",
            "3dsadsadsadsadsa",
            "4dsadsadsadsadsa",
            "5dsadsadsadsadsa",
            "1dsadsadsadsadsa",
            "2dsadsadsadsadsa",
            "3dsadsadsadsadsa",
            "4dsadsadsadsadsa",
            "4dsadsadsadsadsa",
            "5dsadsadsadsadsa",
            "56788dsadsadsadsadsa",
        };
        
        [Benchmark]
        public void TwoWhere()
        {
            Test.Where(s => s.Contains('3')).Where(s=> s.Contains('5')).ToList();

        }
        
        [Benchmark]
        public void OneWhere()
        {
            Test.Where(s => s.Contains('3') && s.Contains('5')).ToList();
        }
```

</details>


```
|   Method |     Mean |   Error |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|--------- |---------:|--------:|--------:|-------:|------:|------:|----------:|
| TwoWhere | 467.1 ns | 3.44 ns | 3.05 ns | 0.0334 |     - |     - |     280 B |
| OneWhere | 363.0 ns | 3.80 ns | 3.17 ns | 0.0162 |     - |     - |     136 B |
```